src/utils/positionUtils.ts

```diff
@@ -389,20 +407,32 @@ export const calcTradeFee = ({
   const takerProportionalFee = Big6Math.div(Big6Math.mul(notional, takerProportionalFeeNumerator), takerFee.scale)
 
   const takerLinearFee = Big6Math.mul(notional, takerFee.linearFee)
-  const subtractiveFee = Big6Math.mul(takerLinearFee, referralFee)
-
-  const marketFee = Big6Math.mul(takerLinearFee - subtractiveFee, marketTakerFee)
-  const tradeFee = subtractiveFee + marketFee
+  const notionalTakerFee = Big6Math.mul(marketTakerFee, notional)
+  // TODO: check tradeFee. Document calls for takerFee + makerFee
+  const tradeFee = notionalTakerFee + takerProportionalFee
+  const subtractiveFee = Big6Math.mul(tradeFee, referralFee)
+  const solverFee = Big6Math.mul(subtractiveFee, intentFeeRate)
+  const marketFee = tradeFee - subtractiveFee
+  const riskFee = Big6Math.mul(marketFee, riskFeeRate)
+
+  const oracleFee = Big6Math.mul(marketFee - riskFee, oracleFeeRate)
+  const protocolFee = marketFee - riskFee - oracleFee
   const feeBasisPoints = !Big6Math.isZero(tradeFee) ? Big6Math.div(tradeFee, notional) : 0n
-  const tradeImpact = takerLinearFee + takerProportionalFee + takerAdiabaticFee - tradeFee
+  // TODO: Check - Old impact calc subtracts trade fee
+  // const tradeImpact = takerLinearFee + takerProportionalFee + takerAdiabaticFee - tradeFee
+  const tradeImpact = takerLinearFee + takerProportionalFee + takerAdiabaticFee
 
   tradeFeeInfo = {
-    tradeFee,
+    tradeFee: tradeFee + marketFee,
+    // TODO: return 0 if there is a solver fee?
     tradeImpact,
     feeBasisPoints,
     proportionalFee: takerProportionalFee,
```

Comment:
should we take this opportunity to cleanup the shape of this return value? I was thinking something like

```
{
  tradeFee: { total, basisPoints }
  tradeImpact:  { total, basisPoints, components: { linear, proportional, adiabatic }
}
```

We could eventually add the tradeFee components as well